chasm/lib/activity/activity.go

```diff
@@ -82,18 +105,109 @@ func NewEmbeddedActivity(
 ) {
 }
 
-func (a *Activity) PopulateRecordActivityTaskStartedResponse(ctx chasm.Context, res *historyservice.RecordActivityTaskStartedResponse) error {
+func (a *Activity) createAddActivityTaskRequest(_ chasm.Context, activityRef chasm.ComponentRef) (*matchingservice.AddActivityTaskRequest, error) {
+	protoRef, err := chasm.ComponentRefToProtoRef(activityRef, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// Note: No need to set the vector clock here, as the components track version conflicts for read/write
+	return &matchingservice.AddActivityTaskRequest{
+		NamespaceId:            activityRef.NamespaceID,
+		TaskQueue:              a.ActivityOptions.TaskQueue,
+		ScheduleToStartTimeout: a.ActivityOptions.ScheduleToStartTimeout,
+		Priority:               a.Priority,
+		ComponentRef:           protoRef,
+	}, nil
+}
+
+// HandleRecordActivityTaskStarted updates the activity on recording activity task started and populates the response.
+func (a *Activity) HandleRecordActivityTaskStarted(ctx chasm.MutableContext, params RecordActivityTaskStartedParams) (*historyservice.RecordActivityTaskStartedResponse, error) {
+	if err := TransitionStarted.Apply(a, ctx, nil); err != nil {
+		return nil, err
+	}
+
+	attempt, err := a.Attempt.Get(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	attempt.LastStartedTime = timestamppb.New(ctx.Now(a))
+	attempt.LastWorkerIdentity = params.WorkerIdentity
+
+	if versionDirective := params.VersionDirective.GetDeploymentVersion(); versionDirective != nil {
+		attempt.LastDeploymentVersion = &deploymentpb.WorkerDeploymentVersion{
+			BuildId:        versionDirective.GetBuildId(),
+			DeploymentName: versionDirective.GetDeploymentName(),
+		}
+	}
+
 	store, err := a.Store.Get(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if store == nil {
+		return nil, errors.New("activity store is nil")
```

Comment:
I would just assume that this isn't nil and let the code panic, it'll end up in the same outcome. No need to be defensive here.